apollo-gradle-plugin/src/main/groovy/com/apollographql/apollo/gradle/ApolloPlugin.groovy

```diff
@@ -40,11 +36,14 @@ class ApolloPlugin implements Plugin<Project> {
   @Override
   void apply(Project project) {
     this.project = project
-    if (hasAndroidPlugin(project) || hasJavaPlugin(project)) {
+    project.plugins.withId("java-base") {
```

Comment:
Interesting, could you please point to any documentation regarding `java-base`. Is it true that android projects has `java-base` plugins? What about pure kotlin projects?

Reply:
The idea comes from this comment: https://github.com/apollographql/apollo-android/issues/1066#issue-364923607, so from the conversation @xenoterracide had with someone from gradle community. 
That mentions:

> that you may really want to depend on java-base that both java/-library extend

nothing about android plugins, but I checked that (see below).

For `JavaPlugin` and `JavaLibrary` it obviously does the thing:

- `JavaPlugin` (`` `java` ``) - applies `JavaBasePlugin`
https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java#L245

- `JavaLibraryPlugin` (`` `java-library` ``) - applies `JavaPlugin`
https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaLibraryPlugin.java#L53

and for android-specific plugins:
- `AppPlugin` (`` `com.android.application` ``)
- `LibraryPlugin` (`` `com.android.library` ``)
- `InstantAppPlugin` (`` `com.android.instantapp` ``)

all above inherit from android's `BasePlugin` (`com.android.build.gradle.BasePlugin`) which applies `JavaBasePlugin` in `com.android.build.gradle.BasePlugin#configureProject()`  method call

<img width="303" alt="screen shot 2018-11-23 at 06 30 15" src="https://user-images.githubusercontent.com/2171398/48929498-c255e580-eee9-11e8-8915-b06d769a2df2.png">

which is called for all following locations:

<img width="209" alt="screen shot 2018-11-23 at 06 32 46" src="https://user-images.githubusercontent.com/2171398/48929535-0ba63500-eeea-11e8-9987-d5fd26be7b6a.png">


(this is for android gradle plugin 2.3.3 which is currently used in the project - btw. it's quite outdated, latest stable is 3.2.1 at the moment and there're new plugins added)

- `KotlinPlugin` - applies `BaseJavaPlugin`
https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt#L345